### DISCUSSION TITLE
Confetti rain

### DISCUSSION THREAD

**Discussion by AlemTuzlak** [epicweb-dev/epic-stack#206] [Ideas]
My proposal is very minimal in terms of code, but the idea is the following:
- Adding a confetti 🎊 component
- Adding a ```redirectWithConfetti``` method

This would basically work by redirecting a user to the new page and raining down confetti 🎊 over the screen like github does when you do certain actions

**Comment by kentcdodds** [comment]
I love it. I'm not 100% certain I want it in the main template, but maybe. I think [an example](https://github.com/epicweb-dev/epic-stack/blob/main/docs/examples.md) would be a good start for this. Would you be willing to put that together?

**Reply by AlemTuzlak** [reply]
Of course, sick at the moment but will do it asap and create a PR so you can check it out

**Reply by AlemTuzlak** [reply]
@kentcdodds added an example using your link explanation. Let me know what you think, code can be polished maybe but that's the general idea. It should redurect you and throw confetti as soon as you start it up

**Reply by kentcdodds** [reply]
Great! I found it here: https://github.com/AlemTuzlak/epic-stack-confetti I didn't get a PR for the examples page yet.

**Reply by kentcdodds** [reply]
Based on what I see in [this commit](https://github.com/epicweb-dev/epic-stack/commit/173c6c60eb624cf19fa8637f79e5d864a8a1097e), I think it would actually make a lot of sense to have something like this, but for notifications. This is effectively the same thing as that. I wonder if we could build something generic enough (but still simple) to satisfy both? If so, then I'd happily include it. Any ideas?

**Reply by AlemTuzlak** [reply]
Just to be clear, when you say notifications, you mean toast ones? Yes this could be made generic, I can add another commit to make it more generic so you can add notifications as well

**Reply by kentcdodds** [reply]
Yes, like toast notifications. Thanks!

**Reply by AlemTuzlak** [reply]
Will be pushing the generic version tomorrow if all goes well 🎊🍞

**Reply by AlemTuzlak** [reply]
@kentcdodds alright I pushed the changes, feel free to check the generic version now! If you like it I will create a PR

**Reply by kentcdodds** [reply]
Code looks great. Could you make a short video or take a screenshot of the experience?

**Comment by AlemTuzlak** [comment]
[redirect.webm](https://github.com/epicweb-dev/epic-stack/assets/18480956/6d149200-7c66-48df-a5a9-39fea4e5bd46)
@kentcdodds here is a preview

**Reply by kentcdodds** [reply]
I love it. If you're willing to make a PR I'd be happy to merge.

Let's make it do confetti when people finish onboarding. Let's have a toast pop up after creating, updating, or deleting a note.

**Reply by AlemTuzlak** [reply]
@kentcdodds let me just test it out and figure out how to make it work with setting multiple cookies at the same time and I will create a PR when I have tested it and made sure it all works

**Reply by AlemTuzlak** [reply]
Hey Kent, quick question, is there a way to add multiple cookies at the same time with the Set-cookie header, when I redirect with 2x cookies only the first one works

**Reply by kentcdodds** [reply]
I'm not sure why that would be, but redirects should not be 2xx. They should only ever be 3xx.

**Reply by AlemTuzlak** [reply]
Sergio helped me out, will try again tomorrow and create a PR if it works

**Comment by kentcdodds** [comment]
This has been implemented: https://github.com/epicweb-dev/epic-stack/discussions/239
